File: src/main/java/kitchenpos/menus/application/MenuValidator.java
package kitchenpos.menus.application;

import kitchenpos.menus.application.dto.request.MenuProductCreateRequest;
import kitchenpos.menus.domain.MenuGroupRepository;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.NoSuchElementException;
import java.util.Objects;
import java.util.UUID;

@Component
public class MenuValidator {
    private final MenuGroupRepository menuGroupRepository;

    public MenuValidator(MenuGroupRepository menuGroupRepository) {
        this.menuGroupRepository = menuGroupRepository;
    }

    public void validateMenuGroup(UUID menuGroupId) {
        menuGroupRepository.findById(menuGroupId)
                .orElseThrow(NoSuchElementException::new);
    }

    public void validateMenuProduct(List<MenuProductCreateRequest> menuProductRequests) {
        if (Objects.isNull(menuProductRequests) || menuProductRequests.isEmpty()) {
            throw new IllegalArgumentException();
        }
    }
}

File: src/main/java/kitchenpos/eatinorders/tobe/domain/MenuValidator.kt
package kitchenpos.eatinorders.tobe.domain

import kitchenpos.eatinorders.tobe.dto.MenuValidateRequest
import kitchenpos.eatinorders.tobe.infra.CheckMenuBeforeOrder
import org.springframework.stereotype.Component

@Component
class MenuValidator(
    private val checkMenuBeforeOrder: CheckMenuBeforeOrder,
) {
    fun canOrder(
        requests: List<MenuValidateRequest>,
    ) {
        check(checkMenuBeforeOrder.canOrder(requests)) { "해당 메뉴는 주문이 불가능합니다. 관리자에게 문의 부탁드립니다." }
    }
}
